### PR TITLE
Re-write virt.py to use only cache non-volatile values.

### PR DESCRIPTION
**Pull request by mikalstill**

### CODE REVIEW

File: shakenfist/external_api/app.py [hunk 1]
```diff
@@ -195,11 +195,21 @@ def redirect_instance_request(func):
     # Redirect method to the hypervisor hosting the instance
     def wrapper(*args, **kwargs):
         i = kwargs.get('instance_from_db_virt')
-        if i and i.node != config.NODE_NAME:
-            url = 'http://%s:%d%s' % (i.node, config.get('API_PORT'),
+        if not i:
+            return
+
+        placement = db.get_instance_attribute(
```
Comment (mandoonandy) [inline shakenfist/external_api/app.py:201]: Is the plan to make this (and others) a class property in future changes?

File: shakenfist/external_api/app.py [hunk 2]
```diff
@@ -677,22 +694,24 @@ def post(self, name=None, cpus=None, memory=None, network=None, disk=None,
...
+                                 None, None, instance.static_values['uuid'])
                     ipm = IPManager.from_db(netdesc['network_uuid'])
                     if 'address' not in netdesc or not netdesc['address']:
                         netdesc['address'] = ipm.get_random_free_address()
                     else:
                         if not ipm.reserve(netdesc['address']):
                             m = 'failed to reserve an IP on network %s' % (
                                 netdesc['network_uuid'])
-                            db.enqueue_instance_error(instance.uuid, m)
+                            db.enqueue_instance_error(
```
Comment (mandoonandy) [inline shakenfist/external_api/app.py:713]: I feel the code churn could be a lot less by moving the change into the class property function?

**Review by mandoonandy — COMMENTED** [review]
Reiterated previous comments about properties. Happy to merge and leave any changes to the next PR. I think you said something to that end anyway?

**Comment by mikalstill** [comment]
Yes, to keep this small as discussed I'd rather to my remaining homework in a new PR please.

**Review by mandoonandy — APPROVED** [review]
LGTM

**Pull request merged**